.github/workflows/deploy-runners.yml

```diff
@@ -0,0 +1,59 @@
+name: Build & Deploy Runners
+
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - '*beta*'
+
+permissions:
+  packages: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        module: ["runner"]
+
+    steps:
+      - name: get container name
+        run: |
+          echo "containername=${{ matrix.module }}" | sed 's/\//-/g' > $GITHUB_ENV
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Docker meta (${{ matrix.module }})
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ghcr.io/tum-dev/gocast/${{ env.containername }}
+          tags: |
+            type=ref,event=branch
+            type=ref,event=pr
+            type=semver,pattern={{version}}
+            type=sha
+          flavor: |
+            latest=true
+            prefix=
+            suffix=
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Login to DockerHub
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set outputs
+        id: vars
+        run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+      - name: Build and push Docker images (${{ matrix.module }})
+        uses: docker/build-push-action@v4
+        with:
+          context: ./${{ matrix.module }}
+          pull: true
+          push: true
+          build-args: version=${{ github.ref_name }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
```

Comment:
You might want to configure `cache-from` and related parameters.
Here is an example how to do this: https://docs.docker.com/build/ci/github-actions/cache/#cache-mounts

Likely the builds are already really fast (looking at you, rust), but a few seconds for a few lines...